ee/clickhouse/models/test/test_dead_letter_queue.py

```diff
@@ -43,8 +45,18 @@
 def reset_tables():
     sync_execute("TRUNCATE TABLE events_dead_letter_queue")
 
-    # can't truncate table with kafka engine, reading from it will delete the rows
-    sync_execute("SELECT * FROM kafka_events_dead_letter_queue")
+    # We can't truncate a table using Kafka engine but reading from it will delete all the rows
+    truncate_kafka_table_query = "SELECT * FROM kafka_events_dead_letter_queue"
+
+    # Note: ClickHouse version >= 21.12 do not allow direct select for Kafka/RabbitMQ/FileLog engine tables.
+    #       We can pass `stream_like_engine_allow_direct_select` to override this behavior.
+    #       Once we drop support for ClickHouse < 21.12 we can remove this if.
+    version_range = SimpleSpec(">=21.12")
+    current_version = get_clickhouse_version()
+    if current_version in version_range:
+        truncate_kafka_table_query += " SETTINGS stream_like_engine_allow_direct_select = 1"
```

Comment:
sync_execute has a settings argument, use that whenever possible.

Also I believe we can just always pass this setting irregardless of the version

Reply:
> sync_execute has a settings argument, use that whenever possible.

TIL

> Also I believe we can just always pass this setting irregardless of the version

The query will fail for versions < 21.12 as the setting was not present yet

Reply:
Clickhouse indeed supports (ignores) unknown settings.